core-bundle/src/DependencyInjection/ContaoCoreExtension.php

```diff
@@ -137,6 +137,8 @@ public function load(array $configs, ContainerBuilder $container): void
         $container->setParameter('contao.intl.countries', $config['intl']['countries']);
         $container->setParameter('contao.insert_tags.allowed_tags', $config['insert_tags']['allowed_tags']);
         $container->setParameter('contao.sanitizer.allowed_url_protocols', $config['sanitizer']['allowed_url_protocols']);
+        $container->setParameter('env(DNS_MAPPING)', '[]');
```

Comment:
Why do you set `env(DNS_MAPPING)` here?

Reply:
This is the fallback in case the `DNS_MAPPING` environment variable is not set.

Reply:
And why do we need that?

Reply:
Otherwise there will be a fatal error if you did not define the `DNS_MAPPING` environment variable (which is the default state).

Reply:
Just as a side note: this is Symfony standard (the `getParameter` method has a special fallback logic for this).

Btw. another way would be to directly write the following into our `services.yaml`:

```yaml
parameters:
    env(DNS_MAPPING ): []
    contao.dns_mapping: '%env(json:DNS_MAPPING)%'
```

instead of defining it in the `ContaoCoreExtension`.

See also the examples here: https://symfony.com/doc/current/configuration/env_var_processors.html

Reply:
@contao/developers Would you prefer defining this in the `services.yaml` or in the `ContaoCoreExtension` class?

Reply:
Yes, we rejected defining parameters directly in our YAMLs once before.

Reply:
You are right, I remember that.